ceph_deploy/tests/parser/test_install.py

```diff
@@ -0,0 +1,159 @@
+        args = self.parser.parse_args('install host1'.split())
+        # dev defaults to master, but version_kind nullifies it
+        assert args.dev == "master"
+        assert args.version_kind == "stable"
+
+    def test_install_dev_custom_version(self):
+        args = self.parser.parse_args('install --dev v0.80.8 host1'.split())
+        assert args.dev == "v0.80.8"
+        assert args.version_kind == "dev"
+
+    @pytest.mark.skipif(reason="test reflects desire, but not code reality")
+    def test_install_dev_option_default_is_master(self):
+        # I don't think this is the way argparse works.
+        args = self.parser.parse_args('install --dev host1'.split())
+        assert args.dev == "master"
+        assert args.version_kind == "dev"
+
+    def test_install_release_testing_mutex(self, capsys):
+        with pytest.raises(SystemExit):
+            self.parser.parse_args('install --release hammer --testing host1'.split())
+        out, err = capsys.readouterr()
+        assert 'not allowed with argument' in err
+
+    def test_install_release_dev_mutex(self, capsys):
+        with pytest.raises(SystemExit):
+            self.parser.parse_args('install --release hammer --dev master host1'.split())
+        out, err = capsys.readouterr()
+        assert 'not allowed with argument' in err
+
+    def test_install_testing_dev_mutex(self, capsys):
+        with pytest.raises(SystemExit):
+            self.parser.parse_args('install --testing --dev master host1'.split())
+        out, err = capsys.readouterr()
+        assert 'not allowed with argument' in err
+
+    @pytest.mark.parametrize('comp', COMP_FLAGS)
+    def test_install_component_default_is_false(self, comp):
+        args = self.parser.parse_args('install host1'.split())
+        assert getattr(args, 'install_%s' % comp) is False
+
+    @pytest.mark.parametrize('comp', COMP_FLAGS)
+    def test_install_component_true(self, comp):
+        args = self.parser.parse_args(('install --%s host1' % comp).split())
+        assert getattr(args, 'install_%s' % comp) is True
+
+    @pytest.mark.skipif(reason="http://tracker.ceph.com/issues/12147")
+    def test_install_multi_component(self):
+        args = self.parser.parse_args(('install --mon --rgw host1').split())
+        assert args.install_mon
+        assert args.install_rgw
+
+    def test_install_adjust_repos_default_is_true(self):
+        args = self.parser.parse_args('install host1'.split())
+        assert args.adjust_repos
+
+    def test_install_adjust_repos_false(self):
+        args = self.parser.parse_args('install --no-adjust-repos host1'.split())
+        assert not args.adjust_repos
+
+    @pytest.mark.skipif(reason="http://tracker.ceph.com/issues/12147")
+    def test_install_adjust_repos_false_with_custom_release(self):
+        args = self.parser.parse_args('install --release firefly --no-adjust-repos host1'.split())
+        assert args.release == "firefly"
+        assert not args.adjust_repos
```

Comment:
why `assert not` instead of asserting the positive ? (not sure what args.adjust_repos is, False?)